### PR TITLE
Create transactions with signatures

### PR DESCRIPTION
**Pull request by Senyoret1**
This PR adds a new function for creating transactions, which not only receives the inputs and outputs, but also the list of signatures. It was added to create transactions signed with the hardware wallet.

### CODE REVIEW

File: liteclient/client.go [hunk 1]
```diff
@@ -108,14 +138,29 @@ func PrepareTransaction(inputsBody string, outputsBody string) string {
 		newTransaction.PushOutput(addr, out.Coins, out.Hours)
 	}
 
-	newTransaction.SignInputs(keys)
+	if signatureList == nil {
+		newTransaction.SignInputs(keys)
+	} else {
+		newTransaction.Sigs = make([]cipher.Sig, len(signatureList))
+		for i, sig := range signatureList {
+			decodedSig, e := hex.DecodeString(sig)
```
Comment (gz-c) [inline liteclient/client.go]: cipher.MustSigFromHex can be used

File: liteclient/client.go [hunk 2]
```diff
@@ -108,14 +138,29 @@ func PrepareTransaction(inputsBody string, outputsBody string) string {
...
+			}
+
+			s := cipher.Sig{}
+			if len(decodedSig) != len(s) {
+				log.Panic("Invalid signature length")
+			}
+			copy(s[:], decodedSig[:])
+			newTransaction.Sigs[i] = s
+		}
+	}
```
Comment (gz-c) [inline liteclient/client.go:146]: Should this panic if len(sigs) != len(txnInputs)?

Comment (Senyoret1) [reply liteclient/client.go:146]: That is done a few lines below with
```
if err := newTransaction.Verify(); err != nil {
    panic(err)
}
```
`newTransaction.Verify()` checks that in
https://github.com/skycoin/skycoin-lite/blob/67bbce84d181c45321ef57cfb4fcb03cca780198/vendor/github.com/skycoin/skycoin/src/coin/transactions.go#L78

File: liteclient/client.go (hunk 1 above)
Comment (Senyoret1) [reply liteclient/client.go]: You're right, that way the code is much cleaner. Change made

File: liteclient/client.go [hunk 3]
```diff
@@ -81,17 +107,20 @@ func PrepareTransaction(inputsBody string, outputsBody string) string {
 	keys := make([]cipher.SecKey, len(inputs))
 
 	for i, in := range inputs {
-		k, err := cipher.SecKeyFromHex(in.Secret)
-		if err != nil {
-			panic(err)
+		if signatureList == nil {
```
Comment (gz-c) [inline liteclient/client.go]: Should it check `len(signatureList) == 0` here instead?

File: liteclient/client.go [hunk 4]
```diff
@@ -65,6 +65,32 @@ func GenerateAddresses(seed string, num int) []Address {
...
+	newTransaction := buildTransaction(inputsBody, outputsBody, signatures)
+	d := newTransaction.Serialize()
+
+	return hex.EncodeToString(d)
+}
+
+// Creates a coin.Transaction using the given lists of inputs, outputs and signatures. If signatureList is nill, the
+// signatures are created using the Secret property of each input.
+// inputsBody and outputsBody are JSONified arrays of TransactionInput and TransactionOutput, respectively.
+// signatureList is a JSONified array of strings.
```
Comment (gz-c) [inline liteclient/client.go]: `signatureList is a JSONified array of strings.` its parsed at this point isn't it?

File: liteclient/client.go [hunk 5]
```diff
@@ -108,14 +137,19 @@ func PrepareTransaction(inputsBody string, outputsBody string) string {
 		newTransaction.PushOutput(addr, out.Coins, out.Hours)
 	}
 
-	newTransaction.SignInputs(keys)
+	if signatureList == nil {
```
Comment (gz-c) [inline liteclient/client.go]: Same `len(signatureList) == 0` comment, generally that is better for checking arrays in go, rather than relying on nil

**Review by gz-c — CHANGES_REQUESTED** [review]

File: liteclient/client.go (hunk 4 above)
Comment (Senyoret1) [reply liteclient/client.go]: Yes, it is. Changed

File: liteclient/client.go (hunk 5 above)
Comment (Senyoret1) [reply liteclient/client.go]: Changed on both places

**Review by gz-c — APPROVED** [review]

**Pull request merged**